### PR TITLE
Improvements for vpc-shared-eni plugin on Linux

### PR DESCRIPTION
**Pull request by ofiliz**
*Issue #, if available:*

*Description of changes:*
This PR consists of two independent commits for vpc-shared-eni Linux plugin:
1) Add capability to configure a neighbor entry for gateway IP address in pod network namespace. This will be used by L3 networking solution in a future PR.
2) Add a dummy interface to bridge to stabilize its MAC address. Comments in the code explain why this is necessary. Also making deleteBridge failure path best-effort by continuing on "does not exist" errors.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: plugins/vpc-shared-eni/network/bridge_linux.go [hunk 1]
```diff
@@ -596,6 +642,7 @@ func (nb *BridgeBuilder) setupVethLink(
 		route := &netlink.Route{
 			LinkIndex: iface.Index,
 			Gw:        gatewayIPAddress,
+			Flags:     int(netlink.FLAG_ONLINK),
```
Comment (mogren) [inline plugins/vpc-shared-eni/network/bridge_linux.go]: Hmm, this sets `RTNH_F_ONLINK` that is used in netlink.c [here](https://github.com/nixnodes/bird/blob/master/sysdep/linux/netlink.c#L368) and [here](https://github.com/nixnodes/bird/blob/master/sysdep/linux/netlink.c#L868), and that toggles masking with `IADDR_SCOPE_MASK` in [neighbor.c](https://github.com/nixnodes/bird/blob/master/nest/neighbor.c#L150). Why is this needed? What does it do?

Comment (ofiliz) [reply plugins/vpc-shared-eni/network/bridge_linux.go]: It asks the IP layer to pretend the next hop is on this link, even when the destination IP address subnet prefix and the source IP address subnet prefix does not match. This is necessary for the following scenario: When we assign an IP address from a CIDR block (as opposed to a real VPC IP address) to a pod, the pod's IP address might look something like 10.10.10.5/24 whereas the ENI's VPC subnet might look like 192.168.0.0/16. When the pod tries to reach the subnet default gateway 192.168.0.1, it will see that it's source address 10.10.10.5 is not in the same subnet as the gateway. Normally, this would cause the IP layer to return a "no route to destination" because it can't reach the default gateway directly. The onlink flag says that the gateway 192.168.0.1 is actually on the same link, and the pod can reach it simply by resolving that IP to a MAC address and sending the datagram directly to it. We create the neighbor entry here, and tell IP layer with onlink flag on the route that it can talk to the default gateway directly.

Comment (mogren) [reply plugins/vpc-shared-eni/network/bridge_linux.go]: Thanks a lot Onur for the thorough explanation. Makes total sense when you know it, but it's an easy thing to miss.
